### PR TITLE
ISD-1723 Add redeploy docs

### PR DESCRIPTION
**Pull request by Thanhphan1147**
Add documentations to describe the jenkins charm redeployment process

### Checklist

- [x] The [charm style guide](https://juju.is/docs/sdk/styleguide) was applied
- [x] The [contributing guide](https://github.com/canonical/is-charms-contributing-guide) was applied
- [x] The changes are compliant with [ISD054 - Managing Charm Complexity](https://discourse.charmhub.io/t/specification-isd014-managing-charm-complexity/11619)
- [x] The documentation is generated using `src-docs`
- [x] The documentation for charmhub is updated.
- [x] The PR is tagged with appropriate label (`urgent`, `trivial`, `complex`)

### CODE REVIEW

File: docs/how-to/redeploy.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# How to redeploy Jenkins
+
+We consider a redeployment is the process where the old charm instance is removed and data is migrated to a new charm instance.
```
Comment (varshigupta12) [inline docs/how-to/redeploy.md:3]: Nit: "Redeployment is a process where the old charm instance is removed and data is migrated to a new charm instance."

File: docs/how-to/redeploy.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+As Jenkins does not use any persistent storage or database, redeploying the charm consists of 3 steps:
```
Comment (jdkandersson) [inline docs/how-to/redeploy.md:5]: Don't we have a volume that we attach for configuration, plugins etc which is persistent storage?

File: docs/how-to/redeploy.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+```
+2. Migrate Jenkins data
+See the `Migrate Jenkins data` section below.
+3. Remove the old Jenkins charm instance
+```bash
+juju remove-application jenkins-k8s
+```
+
+### Migrate Jenkins data
+Follow the instructions in [the charm's documentation for resizing storage](https://charmhub.io/jenkins-k8s/docs/how-to-resize-jenkins-storage) to migrate the data to the new Jenkins charm instance.
```
Comment (mthaddon) [inline docs/how-to/redeploy.md:19]: As discussed on a call just now, let's move the "backup and restore" parts of that doc to its own how-to and then refer to that from here and also from the how-to-resize-jenkins-storage. We already have a ticket for creating an action for those steps as well, but that's outside of the scope of this change.

File: docs/how-to/redeploy.md (hunk 2 above)
Comment (Thanhphan1147) [reply docs/how-to/redeploy.md:5]: Yes, I put `persistent` as a mistake, I'll change it to "As Jenkins does not use a database"

File: docs/how-to/redeploy.md (hunk 1 above)
Comment (Thanhphan1147) [reply docs/how-to/redeploy.md:3]: Thanks, that was my 2 lines of thought joining and I forgot to clean them up.

File: docs/how-to/redeploy.md (hunk 3 above)
Comment (Thanhphan1147) [reply docs/how-to/redeploy.md:19]: will do, thanks, this will indeed make it more straightforward

Comment (Thanhphan1147) [reply docs/how-to/redeploy.md]: I updated the docs to reflect this

File: docs/how-to/redeploy.md (hunk 2 above)
Comment (Thanhphan1147) [reply docs/how-to/redeploy.md]: In the end I thought the first part is not necessary, so I removed it entirely.

File: docs/how-to/redeploy.md (hunk 1 above)
Comment (Thanhphan1147) [reply docs/how-to/redeploy.md]: I updated it

**Review by jdkandersson — DISMISSED** [review]

File: docs/how-to/backup-and-restore-jenkins.md
Comment (yanksyoon) [inline docs/how-to/backup-and-restore-jenkins.md:1]: Would you be able to update this with the latest instructions please? I see that some changes that I've made (e.g. chown -R jenkins:jenkins) isn't reflected here :)

**Review by yanksyoon — COMMENTED** [review]
Minor comment, would you be able to pull the backup-and-restore-jenkins from the latest edits please?

File: docs/how-to/backup-and-restore-jenkins.md
Comment (Thanhphan1147) [reply docs/how-to/backup-and-restore-jenkins.md:1]: got it, I brought over your latest changes to this one.

**Review by codethulu — APPROVED** [review]

**Comment by Thanhphan1147** [comment]
Force merging this since we've decided that since this is a docs PR, it's relatively safe to do so.

**Pull request merged**